examples/facility.rs

```diff
@@ -1,103 +1,133 @@
+                .collect()
+        })
+        .collect();
+
+    let expr = Zip::new((
+        open.iter().chain(Itertools::flatten(transport.iter())),
+        fixed_costs
+            .iter()
+            .chain(Itertools::flatten(trans_costs.iter())),
+    ))
     .fold(LinExpr::new(), |expr, (x, &c)| expr + c * x);
-  model.set_objective(expr, Minimize).unwrap();
-  model.update().unwrap();
-
-  for (p, (&capacity, open)) in Zip::new((&capacity, &open)).enumerate() {
-    let lhs = transport.iter().map(|t| &t[p]).fold(LinExpr::new(), |expr, t| expr + t);
-    model.add_constr(&format!("Capacity{}", p), lhs - capacity * open, Less, 0.0).unwrap();
-  }
-
-  for (w, (&demand, transport)) in Zip::new((&demand, &transport)).enumerate() {
-    let lhs = transport.iter().fold(LinExpr::new(), |expr, t| expr + t);
-    model.add_constr(&format!("Demand{}", w), lhs, Equal, demand).unwrap();
-  }
-
-  for o in open.iter() {
-    o.set(&mut model, attr::Start, 1.0).unwrap();
-  }
-
-  println!("Initial guesss:");
-  let max_fixed = fixed_costs.iter().cloned().fold(-1. / 0., f64::max);
-  for (p, (open, &cost)) in Zip::new((&open, &fixed_costs)).enumerate() {
-    if cost == max_fixed {
-      open.set(&mut model, attr::Start, 0.0).unwrap();
-      println!("Closing plant {}", p);
-      break;
+    model.set_objective(expr, Minimize).unwrap();
+    model.update().unwrap();
+
+    for (p, (&capacity, open)) in Zip::new((&capacity, &open)).enumerate() {
+        let lhs = transport
+            .iter()
+            .map(|t| &t[p])
+            .fold(LinExpr::new(), |expr, t| expr + t);
+        model
+            .add_constr(&format!("Capacity{}", p), lhs - capacity * open, Less, 0.0)
+            .unwrap();
+    }
+
+    for (w, (&demand, transport)) in Zip::new((&demand, &transport)).enumerate() {
+        let lhs = transport.iter().fold(LinExpr::new(), |expr, t| expr + t);
+        model
+            .add_constr(&format!("Demand{}", w), lhs, Equal, demand)
+            .unwrap();
+    }
+
+    for o in &open {
+        o.set(&mut model, attr::Start, 1.0).unwrap();
     }
-  }
-  println!("");
-
-  // use barrier to solve root relaxation.
-  model.get_env_mut().set(param::Method, 2).unwrap();
-
-  // solve.
-  model.optimize().unwrap();
-
-  // print solution.
-  println!("\nTOTAL COSTS: {}", model.get(attr::ObjVal).unwrap());
-  println!("SOLUTION:");
-  for (p, open) in open.iter().enumerate() {
-    let x = open.get(&model, attr::X).unwrap();
-    if x == 1.0 {
-      println!("Plant {} is open", p);
-      for (w, trans) in transport.iter().enumerate() {
-        let t = trans[p].get(&model, attr::X).unwrap();
-        if t > 0.0 {
-          println!("  Transport {} units to warehouse {}", t, w);
+
+    println!("Initial guess:");
+    let max_fixed = fixed_costs.iter().cloned().fold(-1. / 0., f64::max);
+    for (p, (open, &cost)) in Zip::new((&open, &fixed_costs)).enumerate() {
+        if (cost - max_fixed).abs() < f64::EPSILON {
```

Comment:
This should remain an exact comparison. The `max_fixed` variable is one of the values in `fixed_costs`, so the exact comparison is appropriate.